### ISSUE TITLE
 CommissioningController.commissionNode() never returns in some specific cases

### ISSUE THREAD

**Issue by spudwebb** [project-chip/matter.js#1365]
If BLE is enabled in the options, but no actual BLE interface is present on the system, and if the device to commission is not there (e.g not powered on), then the `CommissioningController.commissionNode()` never returns it gets stuck in `NobleBleClient.stopScanning()`.
In this scenario I expected commissionNode() to throw a CommissioningError saying no devices have been discovered, just like it does when BLE is not enabled in the options.

I can replicate the problem with both my own controller and the matter shell:

without BLE enabled, an error is displayed
```
matter-node> commission pair --pairingCode 33068855906
{"discovery":{"knownAddress":"undefined","identifierData":{"shortDiscriminator":13},"discoveryCapabilities":{"ble":false,"onIpNetwork":true}},"passcode":91600864,"commissioning":{"nodeId":"undefined","regulatoryLocation":1,"regulatoryCountryCode":"XX"}}
2024-11-05 15:03:43.148 INFO   MatterController     Commissioning device with identifier {"shortDiscriminator":13} and 1 scanners and knownAddress "undefined"
2024-11-05 15:03:43.148 INFO   ControllerDiscovery  Start Discovering devices using identifier {"shortDiscriminator":13} ...
2024-11-05 15:04:13.147 INFO   ControllerDiscovery  Found 0 devices using identifier {"shortDiscriminator":13}
Error happened during command: Error: No device discovered using identifier {"shortDiscriminator":13}! Please check that the relevant device is online.
Error: No device discovered using identifier {"shortDiscriminator":13}! Please check that the relevant device is online.
    at C:\temp\matter-shell\node_modules\@matter.js\protocol\dist\cjs\protocol\ControllerDiscovery.js:51:15
```

with BLE enabled, no error displayed and the command never returns
```
matter-node> commission pair --ble --pairingCode 33068855906
{"discovery":{"knownAddress":"undefined","identifierData":{"shortDiscriminator":13},"discoveryCapabilities":{"ble":true,"onIpNetwork":true}},"passcode":91600864,"commissioning":{"nodeId":"undefined","regulatoryLocation":1,"regulatoryCountryCode":"XX"}}
2024-11-05 15:02:08.682 INFO   MatterController     Commissioning device with identifier {"shortDiscriminator":13} and 2 scanners and knownAddress "undefined"
2024-11-05 15:02:08.682 INFO   ControllerDiscovery  Start Discovering devices using identifier {"shortDiscriminator":13} ...
2024-11-05 15:02:38.684 INFO   ControllerDiscovery  Found 0 devices using identifier {"shortDiscriminator":13}
```

I guess we should not call the BLE stack to stop scanning, if the BLE scan was never started in the first place.

**Comment by Apollon77** [comment]
Can you please post Debug log from both cases to compare. Thank you

**Comment by spudwebb** [comment]
Here it is:
```
matter-node> commission pair --ble --pairingCode 33068855906
{"discovery":{"knownAddress":"undefined","identifierData":{"shortDiscriminator":13},"discoveryCapabilities":{"ble":true,"onIpNetwork":true}},"passcode":91600864,"commissioning":{"nodeId":"undefined","regulatoryLocation":1,"regulatoryCountryCode":"XX"}}
2024-11-05 17:35:43.376 INFO   MatterController     Commissioning device with identifier {"shortDiscriminator":13} and 2 scanners and knownAddress "undefined"
2024-11-05 17:35:43.377 INFO   ControllerDiscovery  Start Discovering devices using identifier {"shortDiscriminator":13} ...
2024-11-05 17:35:43.377 DEBUG  MdnsScanner          Registered waiter for query _S13._sub._matterc._udp.local with timeout 30 seconds
2024-11-05 17:35:43.378 DEBUG  NobleBleClient       noble state is not poweredOn ... delay scanning till poweredOn
2024-11-05 17:35:43.379 DEBUG  MdnsScanner          Set 2 query records for query _S13._sub._matterc._udp.local: [{"name":"_matterc._udp.local","recordClass":1,"recordType":12},{"name":"_S13._sub._matterc._udp.local","recordClass":1,"recordType":12}]
2024-11-05 17:35:43.379 DEBUG  BleScanner           Registered waiter for query SD:13 with timeout 30 seconds
2024-11-05 17:35:43.379 DEBUG  MdnsScanner          Sending 2 query records for 1 queries with 0 known answers. Re-Announce in 1.5 seconds
2024-11-05 17:35:43.394 DEBUG  UdpChannelNode       Socket created and bound remoteAddress: undefined:5353 localAddress: 0.0.0.0:5353
2024-11-05 17:35:43.395 DEBUG  UdpChannelNode       Socket created and bound remoteAddress: undefined:5353 localAddress: :::5353
2024-11-05 17:35:43.395 DEBUG  UdpChannelNode       Socket created and bound remoteAddress: undefined:5353 localAddress: 0.0.0.0:5353
2024-11-05 17:35:43.395 DEBUG  UdpChannelNode       Socket created and bound remoteAddress: undefined:5353 localAddress: :::5353
2024-11-05 17:35:43.396 DEBUG  UdpChannelNode       Socket created and bound remoteAddress: undefined:5353 localAddress: 0.0.0.0:5353
2024-11-05 17:35:43.396 DEBUG  UdpChannelNode       Socket created and bound remoteAddress: undefined:5353 localAddress: :::5353
2024-11-05 17:35:43.402 DEBUG  UdpChannelNode       Initialize multicast address: 192.168.1.159:5353 interface: Ethernet type: udp4
2024-11-05 17:35:43.406 DEBUG  UdpChannelNode       Initialize multicast address: ::%5:5353 interface: Ethernet type: udp6
2024-11-05 17:35:43.413 DEBUG  UdpChannelNode       Initialize multicast address: 192.168.56.1:5353 interface: VirtualBox Host-Only Network #2 type: udp4
2024-11-05 17:35:43.416 DEBUG  UdpChannelNode       Initialize multicast address: ::%13:5353 interface: VirtualBox Host-Only Network #2 type: udp6
2024-11-05 17:35:43.423 DEBUG  UdpChannelNode       Initialize multicast address: 172.24.0.1:5353 interface: vEthernet (WSL) type: udp4
2024-11-05 17:35:43.427 DEBUG  UdpChannelNode       Initialize multicast address: ::%28:5353 interface: vEthernet (WSL) type: udp6
2024-11-05 17:35:44.887 DEBUG  MdnsScanner          Sending 2 query records for 1 queries with 0 known answers. Re-Announce in 3 seconds
2024-11-05 17:35:47.889 DEBUG  MdnsScanner          Sending 2 query records for 1 queries with 0 known answers. Re-Announce in 6 seconds
2024-11-05 17:35:53.888 DEBUG  MdnsScanner          Sending 2 query records for 1 queries with 0 known answers. Re-Announce in 12 seconds
2024-11-05 17:36:05.896 DEBUG  MdnsScanner          Sending 2 query records for 1 queries with 0 known answers. Re-Announce in 24 seconds
2024-11-05 17:36:13.386 DEBUG  MdnsScanner          Finishing waiter for query _S13._sub._matterc._udp.local, resolving: true
2024-11-05 17:36:13.388 DEBUG  MdnsScanner          Removing last query _S13._sub._matterc._udp.local and stopping announce timer
2024-11-05 17:36:13.389 INFO   ControllerDiscovery  Found 0 devices using identifier {"shortDiscriminator":13}
2024-11-05 17:36:13.392 DEBUG  BleScanner           Finishing waiter for query SD:13, resolving: true
2024-11-05 17:36:13.393 DEBUG  NobleBleClient       Stop BLE scanning for Matter Services ...
```

**Comment by Apollon77** [comment]
If you like try https://github.com/project-chip/matter.js/pull/1366

**Comment by spudwebb** [comment]
I tested it with 0.11.3, and I confirm it works as expected now. Thanks!

**Comment by Apollon77** [comment]
Thansk a lot. Thats also released as 0.11.3 since yesterday
